updatecli/scripts/check-jdk.sh

```diff
@@ -30,10 +30,10 @@ function get_jdk_download_url() {
       echo "https://github.com/adoptium/temurin19-binaries/releases/download/jdk-${jdk_version}/OpenJDK19U-jdk_${platform}_hotspot_${jdk_version//+/_}";
       return 0;;
     21*)
-      # JDK version (21+35-ea-beta)
-      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35-ea-beta/OpenJDK21U-jdk_aarch64_linux_hotspot_ea_21-0-35.tar.gz
+      # JDK version (21+35)
+      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-jdk_aarch64_linux_hotspot_21_35.tar.gz
       urlEncodedJDKVersion="${jdk_version//+/%2B}"
-      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${urlEncodedJDKVersion}/OpenJDK21U-jdk_${platform}_hotspot_ea_21-0-$(echo ${jdk_version} | cut -d '+' -f 2 | cut -d '-' -f 1)"
+      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${urlEncodedJDKVersion}/OpenJDK21U-jdk_${platform}_hotspot_${jdk_version//+/_}"
```

Comment:
According to https://github.com/adoptium/temurin21-binaries/releases/tag/jdk-21%2B35, the links contain OpenJDK21U-**debugimage**_<...> (ex: `https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-debugimage_aarch64_alpine-linux_hotspot_21_35.tar.gz`)

This will certainly evolve, I would add a TODO about removing it when it will change.

Reply:
nope, the `debugimage` series is only a subset of all the elements. These are not the binary we want (it was the case for previous JDK btw)

For Linux aarch64 (not Alpine) we want the following:

```
https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-jdk_aarch64_linux_hotspot_21_35.tar.gz
```

which means the code in this line looks good

Reply:
Quoting my previous comment:

> Nevermind, yesterday they were the only ones available, didn't check the link again today.